extensions/jms/src/main/java/org/mule/extensions/jms/api/operation/JmsPublish.java

```diff
@@ -0,0 +1,173 @@
+import org.mule.extensions.jms.api.config.JmsProducerConfig;
+import org.mule.extensions.jms.api.connection.JmsConnection;
+import org.mule.extensions.jms.api.connection.JmsSession;
+import org.mule.extensions.jms.api.exception.JmsExtensionException;
+import org.mule.extensions.jms.internal.support.JmsSupport;
+import org.mule.runtime.extension.api.annotation.dsl.xml.XmlHints;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.UseConfig;
+import org.mule.runtime.extension.api.annotation.param.display.Summary;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.jms.DeliveryMode;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.Topic;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Operation that allows the user to send a message to a JMS {@link Destination}
+ *
+ * @since 4.0
+ */
+public class JmsPublish {
+
+  private static final Logger logger = LoggerFactory.getLogger(JmsPublish.class);
+
+
+  /**
+   * Operation that allows the user to send a {@link Message} to a JMS {@link Destination
+   *
+   * @param config the current {@link JmsProducerConfig}
+   * @param connection the current {@link JmsConnection}
+   * @param destination the name of the {@link Destination} where the {@link Message} should be sent
+   * @param isTopic {@code true} if the {@code destination} is a {@link Topic}
+   * @param messageBuilder the {@link MessageBuilder} used to create the {@link Message} to be sent
+   * @param persistentDelivery {@code true} if {@link DeliveryMode#PERSISTENT} should be used
+   * @param priority the {@link Message#getJMSPriority} to be set
+   * @param timeToLive the time the message will be in the broker before it expires and is discarded
+   * @param timeToLiveUnit unit to be used in the timeToLive configurations
+   * @param deliveryDelay Only used by JMS 2.0. Sets the delivery delay to be applied in order to postpone the Message delivery
+   * @param deliveryDelayUnit Time unit to be used in the deliveryDelay configurations
+   * @throws JmsExtensionException if an error occurs
+   */
+  public void publish(@UseConfig JmsProducerConfig config, @Connection JmsConnection connection,
+                      @XmlHints(
+                          allowReferences = false) @Summary("The name of the Destination where the Message should be sent") String destination,
+                      @Optional(defaultValue = "false") @Summary("The kind of the Destination") boolean isTopic,
+                      @Summary("A builder for the message that will be published") MessageBuilder messageBuilder,
+                      @Optional @Summary("If true, the Message will be sent using the PERSISTENT JMSDeliveryMode") Boolean persistentDelivery,
```

Comment:
this is nullable on purpose, so we can resolve the override with the config if not set at operation level